tests/e2e/test_e2e.py

```diff
@@ -35,50 +35,51 @@
 
 
 # Set of helper functions to make parametrization less verbose.
-def regression(model):
+def regression(model, test_fraction=0.02):
     return (
         model,
-        utils.get_regression_model_trainer(),
+        utils.get_regression_model_trainer(test_fraction),
         REGRESSION,
     )
 
 
-def classification(model):
+def classification(model, test_fraction=0.02):
     return (
         model,
-        utils.get_classification_model_trainer(),
+        utils.get_classification_model_trainer(test_fraction),
         CLASSIFICATION,
     )
 
 
-def classification_binary(model):
+def classification_binary(model, test_fraction=0.02):
     return (
         model,
-        utils.get_binary_classification_model_trainer(),
+        utils.get_binary_classification_model_trainer(test_fraction),
         CLASSIFICATION,
     )
 
 
-def regression_random(model):
+def regression_random(model, test_fraction=0.02):
```

Comment:
Does `test_fraction` increase for random datasets allow to reproduce the original issue?

Reply:
Yes, unfortunately the default fraction produced way too few samples to be able to reproduce the issue reliably.

Reply:
Got it! We definitely need to refactor testing routines to be more tunable, e.g. allow to adjust `test_fraction` according to programming language. Refer to https://github.com/BayesWitnesses/m2cgen/pull/114#issuecomment-566822606.